### PR TITLE
Cleanup: add missing braces to if/for statements

### PR DESCRIPTION
**Pull request by nashif**
- arch: arm64: add mising braces to single line if statements
- arch: mips: add mising braces to single line if statements
- drivers: add mising braces to single line if statements
- lib: add mising braces to single line if statements
- samples: add mising braces to single line if statements
- soc: add mising braces to single line if statements
- net: add mising braces to single line if statements
- Bluetooth: controller: add mising braces to single line if statements
- debug: coredump: add mising braces to single line if statements
- fs/testsuite: add mising braces to single line if statements
- tests: add mising braces to single line if statements

### CODE REVIEW

File: drivers/ethernet/eth_w5500.c [hunk 1]
```diff
@@ -347,9 +347,10 @@ static int w5500_set_config(const struct device *dev,
 	if (IS_ENABLED(CONFIG_NET_PROMISCUOUS_MODE) &&
 	    type == ETHERNET_CONFIG_TYPE_PROMISC_MODE) {
 		if (config->promisc_mode) {
-			if (!(mode & BIT(mr)))
+			if (!(mode & BIT(mr))) {
 				return -EALREADY;
 			}
+			}
```
Comment (galak) [inline drivers/ethernet/eth_w5500.c:353]: formatting here seems off

File: lib/posix/getopt/getopt_long.c [hunk 2]
```diff
@@ -223,47 +225,51 @@ parse_long_options(struct getopt_state *state, char * const *nargv,
...
-		else if ((flags & FLAG_LONGONLY) ||
-			 long_options[i].has_arg !=
-			     long_options[match].has_arg ||
-			 long_options[i].flag != long_options[match].flag ||
-			 long_options[i].val != long_options[match].val)
+		} else if ((flags & FLAG_LONGONLY) ||
+			   long_options[i].has_arg != long_options[match].has_arg ||
+			   long_options[i].flag != long_options[match].flag ||
+			   long_options[i].val != long_options[match].val) {
 			second_partial_match = 1;
```
Comment (galak) [inline lib/posix/getopt/getopt_long.c:238]: fix formatting.

File: subsys/jwt/jwt.c [hunk 3]
```diff
@@ -160,7 +160,7 @@ static int jwt_add_header(struct jwt_builder *builder)
 #endif
 	int jwt_header_len = ARRAY_SIZE(jwt_header);
 
-	if (jwt_header_len > builder->len) {
+	if (jwt_header > builder->len) {
```
Comment (galak) [inline subsys/jwt/jwt.c:163]: This seems unrelated??

File: tests/drivers/dma/chan_link_transfer/src/test_dma.c [hunk 4]
```diff
@@ -121,17 +121,19 @@ static int test_task(int minor, int major)
...
-				dma_cfg.source_burst_length) != 0)
+		if (strncmp(tx_data, rx_data2, dma_cfg.source_burst_length) != 0) {
 			return TC_FAIL;
+		}
 	} else if (minor == 1 && major == 0) {
 		/* minor link trigger linked channel except the last one*/
 		if (strncmp(tx_data, rx_data2,
-				dma_block_cfg.block_size - dma_cfg.source_burst_length) != 0)
+			    dma_block_cfg.block_size - dma_cfg.source_burst_length) != 0) {
 			return TC_FAIL;
```
Comment (galak) [inline tests/drivers/dma/chan_link_transfer/src/test_dma.c:131]: fix formatting

File: drivers/ethernet/eth_w5500.c (hunk 1 above)
Comment (nashif) [reply drivers/ethernet/eth_w5500.c:353]: oh man, the fun of applying clang-format only on changed files assuming the original was ok

**Review by galak — CHANGES_REQUESTED** [review]
LGTM, a few places that need formatting fixing, and not sure about the change to jwt.

File: lib/posix/getopt/getopt_long.c (hunk 2 above)
Comment (nashif) [reply lib/posix/getopt/getopt_long.c:238]: what's wrong with this one?

Comment (galak) [reply lib/posix/getopt/getopt_long.c:238]: nothing, looking at some many changes at once, didn't realize that the `long_options` are part of the `if`

File: subsys/jwt/jwt.c (hunk 3 above)
Comment (nashif) [reply subsys/jwt/jwt.c:163]: good catch, no idea why clang-tidy is changing that, so, do not trust clang-tidy and double check.

File: tests/drivers/dma/chan_link_transfer/src/test_dma.c (hunk 4 above)
Comment (galak) [reply tests/drivers/dma/chan_link_transfer/src/test_dma.c:131]: never mind, part of the `if`

Comment (nashif) [reply tests/drivers/dma/chan_link_transfer/src/test_dma.c:131]: this looks fine to me.

**Review by galak — DISMISSED** [review]

**Review by carlocaione — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]
Even the Bluetooth controller had a couple of instances of this :o @cvinayak

**Review by aaillet — DISMISSED** [review]
LGTM, isn't the CI able to check for code guidelines non compliance already ?

**Comment by nashif** [comment]
> LGTM, isn't the CI able to check for code guidelines non compliance already ?

not for this rule....

**Review by cvinayak — DISMISSED** [review]
Bluetooth Controller files approved

**Review by erwango — APPROVED** [review]
\*stm32\* approved

**Review by rerickson1 — APPROVED** [review]
**Modem** changes approved.

**Review by nvlsianpu — APPROVED** [review]
drivers/flash LGTM

**Review by carlocaione — APPROVED** [review]
ARM64 LGTM

**Review by dcpleung — APPROVED** [review]
